### PR TITLE
Df 1968 mobile section dropdown

### PR DESCRIPTION
**Pull request by anselmbradford**
Refactors sidenav to replace drop-down header with "IN THIS SECTION" and maintain page order. Also, carries desktop size item-parent/item-child hierarchy in sidenav to mobile sizes.

## Additions
- Added `.nav-secondary_item-parent` class to visually distinguish browse
  pages from the subpages below them in the sidenav.
- Added horizontal rule class for making an inset rule independent of list item widths.

## Removals
- Apparently unneeded CSS.

## Changes
- Changed `.nav-secondary_subitem` class to `.nav-secondary_item-child`.
- Fixed missing states from `.nav-secondary_link__disabled` class for visited and active links.
- Updated side navigation bar to keep page order at mobile sizes and adds
  "IN THIS SECTION" header text to the navigation bar dropdown menu.

## Testing

- Visit `/the-bureau/` and resize down to mobile size. Should look like screenshots.

## Review

- @sebworks 
- @KimberlyMunoz 
- @jimmynotjim 

## Preview

DESKTOP:
screen shot 2015-06-09 at 4 25 37 pm

screen shot 2015-06-09 at 4 25 30 pm

MOBILE:
screen shot 2015-06-09 at 4 44 05 pm

screen shot 2015-06-09 at 4 44 13 pm

## Notes

- Opened https://github.com/cfpb/cfgov-refresh/issues/635 for the sidenav being below the hero on mobile.
- The dotted line on mobile below "IN THIS SECTION" is just the focus state of clicking the link.
- Note the change to the browse page text on desktop. It's slightly bolder than what's on flapjack.demo. I think this is right, but perhaps @duelj can check.

### CODE REVIEW

**Comment by anselmbradford** [comment]
Pushed edit in https://github.com/cfpb/cfgov-refresh/commit/4e2829d1bfeaeee3cab49e6d5de60a30d481a6c0 to bring back demi to "IN THIS SECTION" header text.

File: src/static/css/nav-secondary.less [hunk 1]
```diff
@@ -9,6 +9,10 @@
 @nav-secondary_link__disabled-outline:         @gray-20;
 @nav-secondary_link__disabled-border:          @nav-secondary_link__disabled-bg;
 
+// TODO: Refactor these variable to use existing CF font size variables.
+@font-size-small: 16px;
+@font-size-large: 18px;
```
Comment (Scotchester) [inline src/static/css/nav-secondary.less]: There are no existing CF font-size variables, other than `@base-font-size-px`. I'm not sure I see the value in creating new variables here.

It's outside the scope of this issue, but I think we need to better articulate our philosophy on units and values and how they're declared.

Comment (anselmbradford) [inline src/static/css/nav-secondary.less]: Yeah agreed. These are not variables I created, but I found them tucked away further down in the file, so I just brought them to the top with the other variables. They should be scrutinized in another PR. They're in use in `unit(@font-size-small / @base-font-size-px, em);`, for example.

File: _includes/sidenav.html [hunk 2]
```diff
@@ -50,11 +46,14 @@ <h1 class="u-visually-hidden">
...
+        <hr class="inset-divider" />
         <ul class="nav-secondary_list">
         {%- for href, id, caption in items %}
-            <li class="{{ 'nav-secondary_item' if loop.index == 1 else 'nav-secondary_subitem' }}">
+            {# TODO: logic around loop.index needs to be updated
+                     so a parent item can appear before first item. #}
+            <li class="nav-secondary_item{{ ' nav-secondary_item-parent' if loop.index == 1 else ' nav-secondary_item-child' }}">
                 <a class="nav-secondary_link{{ ' nav-secondary_link__current' if id == active_item_id }}"
-                   href="{{ href|e }}">
+                   href="{{ href|e if id != active_item_id }}">
```
Comment (Scotchester) [inline _includes/sidenav.html]: Smart!

File: _includes/sidenav.html [hunk 3]
```diff
@@ -50,11 +46,14 @@ <h1 class="u-visually-hidden">
         </button>
     </div>
     <div class="expandable_content">
+        <hr class="inset-divider" />
         <ul class="nav-secondary_list">
         {%- for href, id, caption in items %}
-            <li class="{{ 'nav-secondary_item' if loop.index == 1 else 'nav-secondary_subitem' }}">
+            {# TODO: logic around loop.index needs to be updated
+                     so a parent item can appear before first item. #}
+            <li class="nav-secondary_item{{ ' nav-secondary_item-parent' if loop.index == 1 else ' nav-secondary_item-child' }}">
```
Comment (Scotchester) [inline _includes/sidenav.html]: Not clear on why the parent designation is needed, but is we roll with that, these should be `__parent` and `__child` modifiers, probably, right?

Comment (anselmbradford) [inline _includes/sidenav.html]: Yes, sounds good. The parent designation is needed to target the parent link only. All the items have `nav-secondary_item`, the browse pages will have `nav-secondary_item__parent` and the browse page's subpages will have `nav-secondary_item__child`.

File: _includes/sidenav.html (hunk 2 above)
Comment (anselmbradford) [inline _includes/sidenav.html]: Thanks! I actually realized this should probably remove the whole href attribute on the current link.

File: _includes/sidenav.html (hunk 3 above)
Comment (anselmbradford) [inline _includes/sidenav.html]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/3619ac7ac21e78137539137b4e767f75ed0a7fff

File: _includes/sidenav.html (hunk 2 above)
Comment (Scotchester) [inline _includes/sidenav.html]: Not sure, actually. Having no `href` at all can cause some browsers to remove styling that you might want to remain there.

File: _includes/sidenav.html (hunk 3 above)
Comment (anselmbradford) [inline _includes/sidenav.html]: ...and https://github.com/cfpb/cfgov-refresh/commit/9874073d66bc8d4215d9cbc4763acc401e4edd31

File: _includes/sidenav.html (hunk 2 above)
Comment (anselmbradford) [inline _includes/sidenav.html]: It should be checked in IE I guess http://stackoverflow.com/questions/5637969/is-an-empty-href-valid
Pushed https://github.com/cfpb/cfgov-refresh/commit/013f13a350adff3eb552466f11aeb6f15e689fea but it can be reverted.

Comment (anselmbradford) [inline _includes/sidenav.html]: Checked edit in major evergreen browsers and IE8/9. Looked fine.

**Comment by KimberlyMunoz** [comment]
I gave it a quick test in Android 4.0 and iPhone 8.2 and it looks good. :+1:

**Comment by sebworks** [comment]
:+1:
